tests/misc/CachingTest.py

```diff
@@ -0,0 +1,92 @@
+import unittest
+import os
+import time
+from unittest.mock import patch
+
+from pyprint.NullPrinter import NullPrinter
+
+from coalib.misc.Caching import FileCache
+from coalib.misc.CachingUtilities import pickle_load, pickle_dump
+from coalib.output.printers.LogPrinter import LogPrinter
+from coalib import coala
+
+
+class CachingTest(unittest.TestCase):
+
+    def setUp(self):
+        current_dir = os.path.split(__file__)[0]
+        self.caching_test_dir = os.path.join(
+            current_dir,
+            "caching_testfiles")
+        self.log_printer = LogPrinter(NullPrinter())
+        self.cache = FileCache(self.log_printer, "coala_test", flush_cache=True)
+
+    def test_file_tracking(self):
+        self.cache.track_files({"test.c", "file.py"})
+        self.assertEqual(self.cache.data, {"test.c": -1, "file.py": -1})
+
+        self.cache.untrack_files({"test.c"})
+        self.assertFalse("test.c" in self.cache.data)
+        self.assertTrue("file.py" in self.cache.data)
+
+        self.cache.untrack_files({"test.c", "file.py"})
+        self.assertFalse("test.c" in self.cache.data)
+        self.assertFalse("file.py" in self.cache.data)
+
+    def test_write(self):
+        self.cache.track_files({"test2.c"})
+        self.assertEqual(self.cache.data["test2.c"], -1)
+
+        self.cache.write()
+        self.assertNotEqual(self.cache.data["test2.c"], -1)
+
+    @patch('coalib.misc.Caching.os')
+    def test_get_uncached_files(self, mock_path):
+        file_path = os.path.join(self.caching_test_dir, "test.c")
+        cache = FileCache(self.log_printer, "coala_test3", flush_cache=True)
+
+        # Since this is a new FileCache object, the return must be the full set
+        mock_path.path.getmtime.return_value = 0
+        self.assertEqual(cache.get_uncached_files({file_path}), {file_path})
+
+        cache.track_files({file_path})
+        self.assertEqual(cache.get_uncached_files({file_path}), {file_path})
+
+        cache.write()
+        self.assertEqual(cache.get_uncached_files({file_path}), set())
+
+        # Simulate changing the file and then getting uncached files
+        # Since the file has been edited since the last run it's returned
+        mock_path.path.getmtime.return_value = int(time.time() + 1)
```

Comment:
beep, just return one here, (and then you might also be able to remove the time import)